nncf/quantization/algorithms/post_training/backend.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Dict, Iterable, List, Tuple
+
+from nncf import Dataset
+from nncf.data.dataset import DataItem
+from nncf.quantization.algorithms.post_training.algorithm import TModel
+
+
+class PostTrainingBackend(ABC):
+    @staticmethod
+    @abstractmethod
+    def collect_dataitems_for_children_models(
+        model: TModel, calibration_dataset: Dataset, subset_size: int, model_cnt: int
+    ) -> Iterable[DataItem]:
+        """
+        Returns dataitems for children models of the main model.
+
+        :param model: Model to infer to collect dataitems.
+        :param calibration_dataset: Dataset is used to collect new dataitems.
+        :param subset_size: Size of dataitems to collect
+        :param model_cnt: Global model number.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def make_dataset_for_child_models(dataitems: Iterable[DataItem], backend_params: Dict[str, Any]) -> Dataset:
+        """
+        Return dataset for child models.
+
+        :param dataitems: Data items to collect into dataset.
+        :param backend_params: Backend-specific parameters.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def is_single_model(model: TModel) -> bool:
+        """
+        Chechks whether a model has inner subgraphs to quantize.
+
+        :param model: Model to check.
+        :return: True if the model has no inner subgraphs, otherwise - False.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_child_models(model: TModel) -> List[Tuple[TModel, Dict[str, Any]]]:
```

Comment:
Is it possible to make this method general via using ModelTransformer abstraction?

Reply:
Done